packages/react/src/functions/extract-localized-values.ts

```diff
@@ -0,0 +1,33 @@
+import { traverse } from './traverse';
+
+const isLocalizedField = (value: any) => {
+  return value && typeof value === 'object' && value['@type'] === '@builder.io/core:LocalizedValue';
+};
+
+export const containsLocalizedValues = (data: Record<string, any>) => {
+  if (!data || !Object.getOwnPropertyNames(data).length) {
+    return false;
+  }
+  let hasLocalizedValues = false;
+  traverse(data, value => {
+    if (isLocalizedField(value)) {
+      hasLocalizedValues = true;
+      return;
+    }
+  });
+  return hasLocalizedValues;
+};
+
+export const extractLocalizedValues = (data: Record<string, any>, locale: string) => {
+  if (!data || !Object.getOwnPropertyNames(data).length) {
+    return {};
+  }
+
+  traverse(data, (value, update) => {
+    if (isLocalizedField(value)) {
+      update(value[locale] ?? undefined);
```

Comment:
should this also default to `value.Default` ?

Reply:
yes, I had a discussion with @shyam-builder about this and he was worried that it might cause some confusion showing them content that they wouldn't expect and instead we should be pointing them to use the `locale` prop mandatorily if they are opting in for localization (for this reason we are adding the warning in the editor: https://github.com/BuilderIO/builder-internal/pull/7536) but he'll be able to share more details on this.

Reply:
I see the point, but should match how the API resolve localized fields when it get a localized param, but maybe this should be:
```ts
const defaultValue = typeof locale === undefined ? undefined : value.Default

update(value[locale] ?? defaultValue);

```

Reply:
yeah, that makes sense actually, and we'd also display the warning in our web app for any previously released versions where locale is not passed so I think we're fine. Will update this

Reply:
Thanks @teleaziz i've updated the logic to fallback to "Default", can you please take a look? Also, we'll always have a "Default" key for a localized field right?